#3067: add required info on csv partner export (partial solution)

callmarx opened the pull request:
Partial #3067

### Description
<!-- Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context.
Guide questions:
  - What motivated this change (if not already described in an issue)?
  - What alternative solutions did you consider?
  - What are the tradeoffs for your solution?
   
List any dependencies that are required for this change. (gems, js libraries, etc.)

Include anything else we should know about. -->

This is a partial solution for the issue that includes only the following into the partner agencies export:
  - Agency address (line 1 and line 2)
  - Agency city
  - Agency state
  - Agency zip code
  - website
  - Agency type

As this is my first PR and contact with the project, I intend to make another one to include the information "providing diapers" and "providing period supplies" when I got a better overview of it.

### Type of change

<!-- Please delete options that are not relevant. -->
* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. 
Do we need to do anything else to verify your changes? 
If so, provide instructions (including any relevant configuration) so that we can reproduce? -->

Successful passed on rspec test, which I included into the partner model test, and check it manually the download based on seeds data 

### Screenshots
unnecessary

cielf commented:
Background for whoever is reviewing (probably @dorner) --  @callmarx is new to open source but experienced in Ruby - I suggested doing a partial on this as a Good First Issue since we are out of those and Difficulty-Beginner at the moment.

cielf commented on app/models/partner.rb:
```diff
@@ -203,6 +215,21 @@ def contact_person
+
+    @agency_info = {
+      address: [profile.address1, profile.address2].reject(&:blank?).join(' '),
+      city: profile.city,
```
Another minor functionality quibble -- can we join those two addresses with a comma instead of a space?

cielf commented on spec/models/partner_spec.rb:
```diff
@@ -291,21 +291,39 @@
+      expect(partner.csv_export_attributes).to include(agency_website)
+      expect(partner.csv_export_attributes).to include(agency_type)
       expect(partner.csv_export_attributes).to include(notes)
     end
```
Historical note:  we used to have two separate databases - diaperbase and partnerbase.  partnerbase no longer exists -- the text here probably got missed because of the typo.  We can ditch the "from parnerbase".

cielf commented on app/models/partner.rb:
```diff
@@ -172,6 +172,12 @@ def self.csv_export_headers
       "Agency Name",
       "Agency Email",
+      "Agency address",
+      "Agency City",
```
Minor quibble:  inconsistent capitalization -- please caplitalize address.

cielf commented on spec/models/partner_spec.rb:
```diff
@@ -291,21 +291,39 @@
-                               primary_contact_phone: contact_phone
+                               primary_contact_phone: contact_phone,
+                               address1: agency_address,
+                               city: agency_city,
```
I would do a test with both address1 and address 2.

callmarx commented:
Hi @cielf, I fixed the minor functionality quibbles and implemented a solution to include `other_agency_type` info when `agency_type` is `Partner::AGENCY_TYPES["OTHER"]`

cielf reviewed (commented):
Functionality LGTM.  @dorner  Could you take a quick look to see if there's anything you don't like from a code style pov?

dorner commented on app/models/partner.rb:
```diff
@@ -203,6 +215,21 @@ def contact_person
+    return {} if profile.blank?
+
+    @agency_info = {
+      address: [profile.address1, profile.address2].reject(&:blank?).join(', '),
```
I usually prefer positive over negative statements, so I'd like this to be `.select(&:present?)`

callmarx replied:
it's definitely make the code more readable. Thanks

dorner commented on spec/models/partner_spec.rb:
```diff
@@ -291,21 +291,43 @@
+      expect(partner.csv_export_attributes).to include(agency_state)
+      expect(partner.csv_export_attributes).to include(agency_zipcode)
+      expect(partner.csv_export_attributes).to include(agency_website)
+      expect(partner.csv_export_attributes).to include("#{Partner::AGENCY_TYPES["OTHER"]}: #{other_agency_type}")
```
Isn't the order of columns important? We should be checking the exact array that's returned.

callmarx replied:
Sure! I've changed the test for the exact value of csv_export_attributes and the title to "should has the info in the columns order"

dorner reviewed (changes requested):
More minor quibbles. :smile: Otherwise looks good!

dorner commented:
@callmarx there's a failing test - can you fix it?
